public/app/features/manage-dashboards/DashboardImportPage.tsx

```diff
@@ -22,11 +22,16 @@ interface ConnectedProps {
 interface DispatchProps {
   fetchGcomDashboard: typeof fetchGcomDashboard;
   importDashboardJson: typeof importDashboardJson;
+  resetDashboard: typeof resetDashboard;
 }
 
 type Props = OwnProps & ConnectedProps & DispatchProps;
 
 class DashboardImportUnConnected extends PureComponent<Props> {
+  componentWillUnmount() {
+    this.props.resetDashboard();
```

Comment:
What does this do on non-dashboard pages?

Reply:
What do you mean? :thinking:

Reply:
Was just wondering what `resetDashboard` does. How does it update the state?

Reply:
Find the resetDashboard action could be better named, maybe initImportDashboardPage? And I think it's better to have a cleanUp/reset on unmount rather than reset on mount. doing it on mount will cause it to render one time with old state before state resets. 

For clean up, we have this pattern that can fire a cleanup action on unmount that hugo created
We have a redux pattern for this already in the connectWithCleanUp, https://github.com/grafana/grafana/blob/e4c33e0be8418ae5fc99f4c94ee9bf62f9c1e511/public/app/core/components/connectWithCleanUp.tsx

Reply:
Would still rename this action before merge.  As it does not reset any dashboard